library/CMakeLists.txt

```diff
@@ -139,22 +139,22 @@ if(TS_CPP_AVAILABLE AND SWIG_FOUND AND TARGET_SUPPORTS_SHARED_LIBS)
     if (PYTHONLIBS_FOUND)
       include_directories(${PYTHON_INCLUDE_DIRS})
       set_source_files_properties(tinysplinepython.i PROPERTIES CPLUSPLUS ON)
+      set_source_files_properties(tinysplinepython.i PROPERTIES SWIG_MODULE_NAME tinyspline)
       if (${PYTHONLIBS_VERSION_STRING} MATCHES "^3.")
         message(STATUS "Using Python 3 mode.")
-        set(CMAKE_SWIG_FLAGS -py3 -O)
+        set(CMAKE_SWIG_FLAGS -py3 -O -module tinyspline)
       else()
         message(STATUS "Using Python 2 mode.")
-        set(CMAKE_SWIG_FLAGS -O)
+        set(CMAKE_SWIG_FLAGS -O -module tinyspline)
       endif()
       set(CMAKE_SWIG_OUTDIR ${TS_GENERATED_PYTHON_DIR})
-      swig_add_module(tinysplinepython python tinysplinepython.i tinyspline.c tinysplinecpp.cpp)
-      swig_link_libraries(tinysplinepython ${PYTHON_LIBRARIES})
-      # for some reason the generated make target for python is _tinysplinepython
-      add_custom_command(TARGET _tinysplinepython POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy ${TS_GENERATED_PYTHON_DIR}/tinysplinepython.py
+      swig_add_module(tinyspline python tinysplinepython.i tinyspline.c tinysplinecpp.cpp)      
+      swig_link_libraries(tinyspline ${PYTHON_LIBRARIES})
+      add_custom_command(TARGET _tinyspline POST_BUILD
```

Comment:
I am not sure that I understand what is the intent of this custom command. Do you want to copy the .py file close to the shared library? If this is the intent, it does not work on MSVC, since the library goes under "Debug" or "Release" folders.